### PR TITLE
Improve performance of code caller output handling

### PR DESCRIPTION
**Pull request by nwalters512**
When I implemented similar changes in #6994, I neglected to make them for the container code caller. I also realized that we're doing a lot of unnecessary string allocation here, so I changed things to collect an array of data chunks and only join them all together at the end.

### CODE REVIEW

File: lib/code-caller/code-caller-container.js [hunk 1]
```diff
@@ -418,15 +418,15 @@ class CodeCallerContainer {
...
+    this.outputStdout.push(data);
+    if (data.indexOf('\n') >= 0) {
       this._callIsFinished();
     }
     this.debug('exit _handleStdout()');
   }
   _handleStderr(data) {
     this.debug('enter _handleStderr()');
-    this.outputStderr += data;
+    this.outputStderr.push(data);
```
Comment (mwest1066) [inline lib/code-caller/code-caller-container.js:429]: Where do we do `this.outputStderr.join('')`? I can't see it anywhere.

**Review by mwest1066 — APPROVED** [review]

File: lib/code-caller/code-caller-container.js (hunk 1 above)
Comment (nwalters512) [reply lib/code-caller/code-caller-container.js:429]: We actually never use it at all (I had to double-check). I guess we could remove it entirely, though I like having it around for potential debugging (I can just add `console.log(this.outputStderr.join(''))` to finish handler).

Comment (mwest1066) [reply lib/code-caller/code-caller-container.js:429]: Ok, sounds good, I was just worried we were missing it.

File: lib/code-caller/code-caller-container.js [hunk 2]
```diff
@@ -401,6 +403,8 @@ class CodeCallerContainer {
     this.stderrStream = new MemoryStream();
     this.stdinStream.pipe(stream);
     this.container.modem.demuxStream(stream, this.stdoutStream, this.stderrStream);
+    this.stdoutStream.setEncoding('utf8');
+    this.stderrStream.setEncoding('utf8');
```
Comment (nwalters512) [inline lib/code-caller/code-caller-container.js:407]: This was necessary to ensure we get strings in the handlers, which is consistent with how the native code caller works.

**Pull request merged**